lib/jsdom/living/postmessage.js

```diff
@@ -0,0 +1,25 @@
+"use strict";
+const isValidTargetOrigin = require("../utils").isValidTargetOrigin;
+
+module.exports = function (message, targetOrigin) {
+  let window = this;
+  let event;
+
+  if (!targetOrigin) {
+    throw new TypeError("'postMessage' requires a second argument 'targetOrigin'");
+  }
+
+  if (!isValidTargetOrigin(targetOrigin)) {
+    throw new SyntaxError("Failed to execute 'postMessage' on 'Window': " +
+      "Invalid target origin '" + targetOrigin + "' in a call to 'postMessage'.");
+  }
+
+  if (targetOrigin !== "*" && targetOrigin !== window.origin) {
+    return;
+  }
+
+  event = window.document.createEvent("MessageEvent");
+  event.initEvent("message", false, false);
+  event.data = message;
+  window.dispatchEvent(event);
```

Comment:
Should queue a task for this. I can't remember if we have something for task-queueing already, or whether just `setImmediate` is what you should go with.